src/Microsoft.AspNetCore.Authentication.Core/AuthenticationService.cs

```diff
@@ -0,0 +1,133 @@
+            }
+            return result;
+        }
+
+        public virtual async Task ChallengeAsync(HttpContext httpContext, string authenticationScheme, AuthenticationProperties properties, ChallengeBehavior behavior)
+        {
+            if (authenticationScheme == null)
+            {
+                var defaultChallengeScheme = await Schemes.GetDefaultChallengeSchemeAsync();
+                authenticationScheme = defaultChallengeScheme?.Name;
+                if (authenticationScheme == null)
+                {
+                    throw new InvalidOperationException($"No authenticationScheme was specified, and there was no DefaultChallengeScheme found.");
+                }
+            }
+
+            var handler = await Handlers.GetHandlerAsync(httpContext, authenticationScheme);
+            if (handler == null)
+            {
+                throw new InvalidOperationException($"No authentication handler is configured to handle the scheme: {authenticationScheme}");
+            }
+
+            var challengeContext = new ChallengeContext(httpContext, authenticationScheme, properties, behavior);
+            await handler.ChallengeAsync(challengeContext);
+        }
+
+        public virtual async Task SignInAsync(HttpContext httpContext, string authenticationScheme, ClaimsPrincipal principal, AuthenticationProperties properties)
+        {
+            if (principal == null)
+            {
+                throw new ArgumentNullException(nameof(principal));
+            }
+
+            if (authenticationScheme == null)
+            {
+                var defaultScheme = await Schemes.GetDefaultSignInSchemeAsync();
+                authenticationScheme = defaultScheme?.Name;
+                if (authenticationScheme == null)
+                {
+                    throw new InvalidOperationException($"No authenticationScheme was specified, and there was no DefaultAuthenticateScheme found.");
+                }
+            }
+
+            var handler = await Handlers.GetHandlerAsync(httpContext, authenticationScheme);
+            if (handler == null)
+            {
+                throw new InvalidOperationException($"No authentication handler is configured to handle the scheme: {authenticationScheme}");
+            }
+
+            var signInContext = new SignInContext(httpContext, authenticationScheme, principal, properties);
+            await handler.SignInAsync(signInContext);
+        }
+
+        public virtual async Task SignOutAsync(HttpContext httpContext, string authenticationScheme, AuthenticationProperties properties)
+        {
+            if (string.IsNullOrEmpty(authenticationScheme))
+            {
+                throw new ArgumentException(nameof(authenticationScheme));
+            }
+
+            var handler = await Handlers.GetHandlerAsync(httpContext, authenticationScheme);
```

Comment:
If you're just using forms, social, or WsFed then there's one thing to sign in and out of (cookies).
If you're using Identity then there are several signouts and the app signs out of them all.
If you're using OIDC there are two signouts, and the app needs to choose which.
And if you're using Bearer then signout doesn't work anyways.

So having a default sign out scheme, or even defaulting to the sign in scheme would only help the first scenario which is less common. Forget I asked.